### ISSUE TITLE
support json schema?

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#4588]
I wonder if we should support a JSON schema for our config file.. Apart from the validation rules, it'd be nice for documentation/auto-completion of options. Most IDEs today support that.

Not 100% sure how we'd do this.. Since we have `valhalla_build_config` which is still relevant, even with a schema, we'd probably just install the schema along with everything else and point it there?

Apparently rapidjson even has rudimentary JSON5 support: https://github.com/Tencent/rapidjson/issues/1971. At least comments would be possible. But for Python it's not available in the std lib.

**Comment by kevinkreiser** [comment]
upon reading the title i thought you meant for the http APIs, imho its more useful there where we dont allow breaking the schema (just adding to it).

with the config file we have removed required keys (name changes) and changed values etc. you must be thinking of a specific use-case though.. possibly ops related? or maybe you are cool with the schema changing when we change the config which yeah i guess thats cool too

**Comment by nilsnolde** [comment]
Just popped into my mind tbh :smile: 

No bigger operational concerns, mostly pure convenience. Currently people have to go look inside the sources (or know the key by heart for command line lookup), both of which most people don't do or even know they can do that. A schema would give IDE supported descriptions, auto-completion and input validation (as long as you are in an IDE, otherwise it's still easier to validate against the schema instead of getting some fairly obscure rapidjson/c++ failure). If we version the schema and point to the local version of it by default, we shouldn't have problems with removed/added stuff I think.

A strict CI job could make sure valhalla_build_config adheres to the current schema (and/or vice versa), so we don't forget stuff which is likely if broken out into 2 files (schema and valhalla_build_config).

**Comment by nilsnolde** [comment]
Good point with HTTP API. _If_ we'd give a URI in our response, it could even be self-documenting I guess? (never wrote a JSON schema before myself)

**Comment by ankitkumarpanda14** [comment]
Hey @nilsnolde ,

I am new to this repo, I see it being tagged "good-first-issue". Is it okay if I take up the issue? 

If yes, please do let me know what would be a good starting point for it?

**Comment by Aadityasharma1-programmer** [comment]
Hi, I would like to solve this issue ,if solving this issue is still relevant to the project .Can u assign this to me.
